packages/core/src/components/BrandConcierge/createSendConversationEvent.js

```diff
@@ -0,0 +1,140 @@
+  const sendConversationServiceRequest = createSendConversationServiceRequest({
+    logger,
+    fetch,
+    config,
+  });
+
+  const decodeKndctrCookie = createGetEcidFromCookie({
+    orgId: config.orgId,
+    cookieJar: loggingCookieJar,
+    logger,
+  });
+  return (options) => {
+    let streamingEnabled = false;
+    const { message, onStreamResponse, xdm, data } = options;
+    const sessionId =
+      getConciergeSessionCookie({ loggingCookieJar, config }) || uuid();
+    const payload = createDataCollectionRequestPayload();
+    const request = createConversationServiceRequest({
+      payload,
+      sessionId: sessionId,
+    });
+
+    const event = eventManager.createEvent();
+    if (message || data) {
+      const pageSurface = getPageSurface();
+      event.mergeQuery({
+        conversation: {
+          surfaces: [pageSurface],
+          message,
+          data,
+        },
+      });
+    }
+
+    const ecid = decodeKndctrCookie();
+
+    event.mergeXdm({
+      identityMap: {
+        ECID: [
+          {
+            id: ecid,
+          },
+        ],
+      },
+    });
+
+    event.mergeXdm({ ...xdm });
+
+    if (message || data) {
+      streamingEnabled = true;
+    }
+    const url = buildEndpointUrl({
+      edgeDomain,
+      edgeBasePath,
+      datastreamId,
+      request,
+    });
+    return consent.awaitConsent().then(() => {
+      return lifecycle
+        .onBeforeEvent({
+          event,
+        })
+        .then(() => {
+          try {
+            // NOTE: this calls onBeforeEventSend callback (if configured)
+            event.finalize(onBeforeEventSend);
+          } catch (error) {
+            onStreamResponse({ error });
+            throw error;
+          }
+          payload.addEvent(event);
+          return sendConversationServiceRequest({
+            requestId: uuid(),
+            url,
+            request,
+            onStreamResponse,
+            streamingEnabled,
+          }).then((response) => {
+            if (response.status === 204) {
+              return;
+            }
+            const onStreamResponseCallback = (event) => {
+              if (event.error) {
+                onStreamResponse({ error: event.error });
+              }
+              const substr = event.data.replace("data: ", "");
+              const responseJson = JSON.parse(substr);
+              const response = createResponse({ content: responseJson });
+
+              cookieTransfer.responseToCookies(response);
+
+              logger.info(
+                "onStreamResponse callback called with",
+                response.getPayloadsByType("brand-concierge:conversation"),
+              );
+              onStreamResponse(
+                response.getPayloadsByType("brand-concierge:conversation"),
+              );
+            };
+
+            const streamParser = createStreamParser();
+
+            streamParser(response.body, onStreamResponseCallback);
```

Comment:
you may be able to leverage [`EventSource`](https://developer.mozilla.org/en-US/docs/Web/API/EventSource), which can handle the connection from the server, calling events when a stream response has arrived, and parsing the response.

Reply:
EventSource requires HTTP GET. In a conversational experience user query is submited as a POST request, so we can't use EventSource.